### PR TITLE
NF: refactor DeckOption conf change handler

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
@divyansh-dxn 
While looking at #12592 I realized some code were repeated a lot. I'd love your opinion about this refactoring.
Once this last PR is merged, we can even remove the pre/post parts and just keep a single function there. 

Actually, I realize that if we had done it first, it would have made PR even simpler to review, since in this case, the launchCollectionTask would be replaced by a more or less similar call, with something that does the UI and then inside does the background change

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
By the way @divyansh-dxn do you have any idea why some change in deck options don't trigger a save? That seems a bad idea, even if in the worst case, there is no actual huge dommage

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 1]
```diff
@@ -146,6 +149,20 @@ class DeckOptions :
             return DeckConfig.parseTimerOpt(options, true)
         }
 
+        fun confChangeHandler(timbering: String, block: Collection.() -> Unit) {
```
Comment (ShridharGoel) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:152]: `timbering`? 😅

**Review by ShridharGoel — APPROVED** [review]

**Comment by divyansh-dxn** [comment]
> @divyansh-dxn While looking at #12592 I realized some code were repeated a lot. I'd love your opinion about this refactoring. Once this last PR is merged, we can even remove the pre/post parts and just keep a single function there.
> 
> Actually, I realize that if we had done it first, it would have made PR even simpler to review, since in this case, the launchCollectionTask would be replaced by a more or less similar call, with something that does the UI and then inside does the background change

Yeah probably it was a good idea to extract out it first. But I noticed this when two of the classes were already migrated so on the last one I put a TODO for refactor.
https://github.com/ankidroid/Anki-Android/pull/12592/files#diff-8e63311e69fdb35f97b20c8de1c83019459f4d1a8bd73e4eec0c8a719002f07cR237

**Review by mikehardy — APPROVED** [review]
LGTM

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:152]: I don't understand the joke @ShridharGoel.
 Maybe Timbered?
Too late anyway

Comment (ShridharGoel) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:152]: @Arthur-Milchior It was not a joke, I wasn't sure why you selected that name - so I added the comment. Is "timbering" a combination of "timber" and "string"?

Comment (ShridharGoel) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:152]: I thought you wanted the parameter name to look funny or something, so I wrote the emoji.

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:152]: Oh no, I was just verbing this word. Like, this is what we are timbering.

**Pull request merged**